code/main.py

```
@@ -0,0 +1,122 @@
+# Import required libraries
+import pandas as pd
+import numpy as np
+from sentence_transformers import SentenceTransformer
+from sklearn.cluster import KMeans
+from sklearn.decomposition import PCA
+from tensorflow.keras.models import Model
+from tensorflow.keras.layers import Dense, Input, Embedding, Flatten, Concatenate
```

Ruff (0.8.2)

`tensorflow.keras.layers.Embedding` imported but unused

Remove unused import

(F401)